src/sentry/api/endpoints/release_thresholds/health_checks/is_new_issues_count_healthy.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+from typing import List, Optional
+from uuid import uuid4
+
+from django.db.models import CharField, Count, QuerySet, Value
+
+from sentry.api.endpoints.release_thresholds.types import EnrichedThreshold
+from sentry.models.group import Group
+from sentry.models.release_threshold.constants import ReleaseThresholdType, TriggerType
+
+NEW_ISSUES_COUNT_COLUMN = "count"
+NEW_ISSUES_UNIQUE_KEY_COLUMN = "unique_key"
+
+
+def update_new_issues_count_healthy_for_thresholds(
+    release_thresholds: List[EnrichedThreshold],
+) -> None:
+    """
+    Updates the release thresholds is_healthy for the given iterable.
+    This function leverages bulk querying to update the values of the release thresholds.
+    Each release threshold is considered to be a unique threshold to determine its own health.
+    """
+    queries = {}
+    q: Optional[QuerySet] = None
+    # Loop through each enriched_threshold and get the query that we need to aggregate
+    for ethreshold in release_thresholds:
+        # Ignore any incorrect thresholds that might have slipped in
+        if ethreshold["threshold_type"] != ReleaseThresholdType.NEW_ISSUE_COUNT_STR:
+            continue
+
+        unique_id = str(uuid4())
+        queries[unique_id] = ethreshold
+        annotation = get_is_new_issues_count_healthy_query(ethreshold, unique_id)
+        if q is None:
+            q = annotation
+        else:
+            q = q.union(annotation, all=True)
+
+    if q is None:
+        return
+
+    # Execute the query to retrieve the unique_key and count columns that hold the data we need
+    # Each resultant threshold should be a dictionary with the 2 columns
+    results = q.values(NEW_ISSUES_UNIQUE_KEY_COLUMN, NEW_ISSUES_COUNT_COLUMN)
+
+    # Update the enriched_thresholds with the results
+    for result in results:
+        unique_key = result[
+            NEW_ISSUES_UNIQUE_KEY_COLUMN
+        ]  # Get the unique key from the result dictionary
+        value = result[NEW_ISSUES_COUNT_COLUMN]  # Get the count value
+        ethreshold = queries[unique_key]
```

Comment:
If we need some sort of key to key results off of, we should be able to utilize the threshold id 🤷

Reply:
Yeah we can def re-use release_threshold id